provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DotNetty.Buffers;
+using DotNetty.Codecs.Mqtt.Packets;
+using DotNetty.Transport.Channels;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport.Models;
+using Microsoft.Azure.Devices.Shared;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport
+{
+    internal class ProvisioningChannelHandlerAdapter : ChannelHandlerAdapter
```

Comment:
no liveness support for now?

Reply:
Good point but the comment is a bit ambiguous: please add more information. 
I do see one problem with the cooperative cancellation not being checked if no packets are received. Anything else?

(PS: I do see this is a very good candidate for TLA modeling.)

Reply:
No, I mean MQTT level keep alive handling with pings

Reply:
Provisioning should take no longer than a few seconds. I don't think that would be required in this case.
@simonporter @rajeevmv - Please confirm.

Reply:
Yes it can take longer for a variety of reasons especially in the future when we have out of band attestation, external calls etc.

Reply:
Thanks @rajeevmv @simonporter - I've opened #274 to track this.